services/aws-request.js

```diff
@@ -34,17 +34,28 @@ class AWSService {
         //1. verify token and decode token to get user info
         const userInfo = verifyApiToken(context, config.session_secret);
         //verify submitter
-        await verifySubmitter(userInfo, params?.submissionID, this.submissions, this.userService);
+        const submission = await verifySubmitter(userInfo, params?.submissionID, this.submissions, this.userService);
         //2. create temp credential
         // Initialize an STS object
         const sts = new AWS.STS();
         const timestamp = (new Date()).getTime();
+        //add s3 object access policy
+        const policy = {
+            Version: '2012-10-17',
+            Statement: [
+              {
+                Effect: 'Allow',
+                Action: ['s3:GetObject','s3:PutObject'],
+                Resource: [`arn:aws:s3:::${config.submission_bucket}/${submission.rootPath}/*`]
```

Comment:
Here we should use **submission.bucketName**, so that if we change how bucket for a submission is determined, we don't have to change code here.

Reply:
updated.  Please continue.